### PR TITLE
Feature/discipline

### PR DESCRIPTION
**Pull request by KostyaDedulya**
#### 🤔 This is a ...
- [x] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 🔗 Related issue link

https://github.com/rolling-scopes/rsschool-app/issues/354

#### 💡 Background and solution 

Add disciplines
image

#### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Database migration is added or not needed
- [ ] Documentation is updated/provided or not needed
- [x] Changes are tested locally

### CODE REVIEW

File: server/src/models/discipline.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @CreateDateColumn()
+  createdDate: number;
+
+  @UpdateDateColumn()
+  updatedDate: number;
+
+  @Column({ type: 'text', unique: true })
```
[the remark this answers is not in the document]
Comment (apalchys) [reply server/src/models/discipline.ts:14]: `name` usually is never considered as a unique field (think about name of a person). 
If you want to have some unique field, it's better to consider something more specific. e.g. key
```
{ key: "javascript", name: "JavaScript" },
{ key: "csharp", name: "C#" },
{ key: dotnet", name: ".NET" }
```

**Comment by KostyaDedulya** [comment]
Finally I've done it
Add relations with disciplines and tasks, events

File: client/src/pages/admin/disciplines.tsx [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { getServerSideProps } from '../../modules/Discipline/data/getServerSideProps';
+import { DisciplinePage } from '../../modules/Discipline/pages/DisciplinePage';
+import withSession from '../../components/withSession';
+
+export { getServerSideProps };
+export default withSession(DisciplinePage);
```
Comment (apalchys) [inline client/src/pages/admin/disciplines.tsx:6]: there is intention to remove `withSession` and use SessionContext/Provider.

Please see https://github.com/rolling-scopes/rsschool-app/blob/master/client/src/pages/course/mentor/students.tsx#L8

Comment (apalchys) [reply client/src/pages/admin/disciplines.tsx:6]: https://github.com/rolling-scopes/rsschool-app/blob/master/client/src/modules/Mentor/pages/Students/index.tsx#L14

File: client/src/services/discipline.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+export class DisciplineService {
+  private axios: AxiosInstance;
+
+  constructor(token?: string) {
+    this.axios = axios.create(getServerAxiosProps(token));
+  }
+
+  async getAllDisciplines() {
+    try {
+      const result = await this.axios.get('/api/v2/disciplines');
```
Comment (apalchys) [inline client/src/services/discipline.ts:15]: please use `DisciplineApi` from `api`. We generate client from swagger and don't need to construct urls/services anymore.

**Comment by apalchys** [comment]
Closing in favor https://github.com/rolling-scopes/rsschool-app/pull/1664

I merged conflicts and made some improvements

**Pull request closed without merging**